vision/src/autogluon/vision/detector/detector.py

```diff
@@ -0,0 +1,270 @@
+        """Automatic fit process for object detection.
+
+        Parameters
+        ----------
+        train_data : pd.DataFrame or str
+            Training data, can be a dataframe like image dataset.
+            For more details of how to construct a object detection dataset, please checkout:
+            `http://preview.d2l.ai/d8/main/object_detection/getting_started.html`.
+            If a string is provided, will search for k8 datasets.
+        tuning_data : pd.DataFrame or str, default = None
+            Holdout tuning data for validation, reserved for model selection and hyperparameter-tuning,
+            can be a dataframe like image dataset.
+            If a string is provided, will search for k8 datasets.
+            If `None`, the validation dataset will be randomly split from `train_data` according to `holdout_frac`.
+        time_limit : int, default = None
+            Time limit in seconds, if not specified, will run until all tuning and training finished.
+            If `time_limit` is hit during `fit`, the
+            HPO process will interrupt and return the current best configuration.
+        hyperparameters : dict, default = None
+            Extra hyperparameters for specific models.
+            Accepted args includes(not limited to):
+            epochs : int, default value based on network
+                The `epochs` for model training.
+            batch_size : int
+                Mini batch size
+            lr : float
+                Trainer learning rate for optimization process.
+            You can get the list of accepted hyperparameters in `config.yaml` saved by this predictor.
+        **kwargs :
+            holdout_frac : float, default = 0.1
+                The random split ratio for `tuning_data` if `tuning_data==None`.
+            random_state : int, default = None
+                The random_state(seed) for shuffling data, only used if `tuning_data==None`.
+                Note that the `random_state` only affect the splitting process, not model training.
+                If not specified(None), will leave the original random sampling intact.
+            nthreads_per_trial : int, default = (# cpu cores)
+                Number of CPU threads for each trial, if `None`, will detect the # cores on current instance.
+            ngpus_per_trial : int, default = (# gpus)
+                Number of GPUs to use for each trial, if `None`, will detect the # gpus on current instance.
+            hyperparameter_tune_kwargs: dict, default = None
+                num_trials : int, default = 1
+                    The number of HPO trials when `time_limit` is None. 
+                search_strategy : str, default = 'random'
+                    Searcher strategy for HPO, 'random' by default.
+                    Options include: ‘random’ (random search), ‘bayesopt’ (Gaussian process Bayesian optimization),
+                    ‘grid’ (grid search).
+                max_reward : float, default = 0.9
```

Comment:
Yes, theoretically it's almost impossible to reach 0.9 mean average precision for bonding boxes. Even if a model does, the prediction difference between 0.9 mAP and 1.0 mAP is neglected. 

My input of 0.9 here is to avoid the searcher desperately exploring the best configuration even if the result is already saturated(good for the average users, bad for competition?) What I can do is to put 1.0 back as default, or print a message to tell if the `max_reward` is met but you can possibly improve it.

@Innixma What's you suggestion?

Reply:
Adding `presets` though, where `best_quality` mode will override `max_reward` to 1 though

Reply:
I'll leave this to your decision since you are more familiar with what 0.9 mAP means, but I would suggest that so long as the user specified a long enough time_limit, there is no reason to stop training when improvements in score could still be gained. Perhaps with a higher quality preset it can be set to 1?